### PR TITLE
fix: change dashboard  adaptation on small screen

### PR DESCRIPTION
**Pull request by nuanyang233**
When the screen sizes is smaller than 1184, the drawer will be close. Then menu button is displayed on the left side of icon, click it will open drawer

### CODE REVIEW

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 1]
```diff
@@ -38,57 +41,61 @@ function ListItemLink({ nested, ...props }: LinkProps & ListItemProps & { nested
...
-                </List>
-            </Collapse>
-            <ListItemLink to={Routes.Settings}>
-                <ListItemIcon>
-                    <Settings />
-                </ListItemIcon>
-                <ListItemText primary="Settings" />
-            </ListItemLink>
-        </List>
+        <>
```
Comment (guanbinrui) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:52]: No need add fragment.

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+    primaryAction?: React.ReactNode
+}
 
 const useStyles = makeStyles((theme) => ({
-    root: { backgroundColor: theme.palette.background.paper },
-    temporaryDrawerPaper: {
-        backgroundColor: new Color(theme.palette.background.paper).alpha(0.8).toString(),
+    appBar: {},
+    toolbar: {
+        [theme.breakpoints.up(1184)]: {
```
Comment (guanbinrui) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:28]: Why this magic number? We have MUI breakpoints.

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 3]
```diff
@@ -20,51 +60,89 @@ const useStyles = makeStyles((theme) => ({
...
+                        <Navigation />
+                    </Grid>
+                ) : (
+                    <Drawer
+                        open={open}
+                        onClose={() => setOpen(false)}
+                        BackdropProps={{ invisible: true }}
+                        PaperProps={{ elevation: 0 }}
+                        variant="temporary"
+                        // className={classes.temporaryDrawer}
```
Comment (guanbinrui) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:134]: Remove it.

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 4]
```diff
@@ -1,17 +1,57 @@
...
+            paddingLeft: theme.spacing(1),
+        },
+    },
+    root: {
+        backgroundColor: theme.palette.background.paper,
+        height: `calc(100vh - ${theme.mixins.toolbar.minHeight}px)`,
+    },
+    temporaryDrawer: {
+        width: 232,
+        // Or theme.xxx.width?
```
Comment (guanbinrui) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:41]: Remove it .

File: packages/dashboard/src/components/DashboardFrame/index.tsx (hunk 2 above)
Comment (nuanyang233) [reply packages/dashboard/src/components/DashboardFrame/index.tsx:28]: The small screen on the design is supported by 1184, and I asked the designer if we could set this magic number as our breakpoint if needed

Comment (guanbinrui) [reply packages/dashboard/src/components/DashboardFrame/index.tsx:28]: @xiazhiming

File: packages/dashboard/src/pages/routes.tsx [hunk 5]
```diff
@@ -16,16 +15,15 @@ const Wallets = lazy(() => import('./Wallets'))
 const Welcome = lazy(() => import('./Welcome'))
 const Personas = lazy(() => import('./Personas'))
 const Settings = lazy(() => import('./Settings'))
+
 export function Pages() {
     return (
         <Suspense fallback="loading...">
             <Switch>
                 <Route path={Routes.Welcome} children={<Welcome />} />
-                <DashboardFrame>
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/routes.tsx:24]: If it is arranged this way, the state of `DashboardFrame` will be lost in the transition of different routes. This is why I choose to put it here. Can you resolve this problem in your design?

Comment (nuanyang233) [reply packages/dashboard/src/pages/routes.tsx:24]: I also noticed this problem and the whole root dom tree changed every time I switched routes.I will make some attempts, If I can't resolve, I will add small screen adaptation on your DashboardFrame

File: packages/dashboard/src/components/DashboardFrame/index.tsx (hunk 2 above)
Comment (Jack-Works) [reply packages/dashboard/src/components/DashboardFrame/index.tsx:23]: yeah, use larger breakpoints

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 6]
```diff
@@ -46,20 +117,57 @@ export interface PageFrameProps extends React.PropsWithChildren<{}> {
...
+                        PaperProps={{ elevation: 0 }}
+                        variant="temporary"
+                        ModalProps={{
+                            BackdropProps: {
+                                classes: { root: classes.drawerBackdropRoot },
+                            },
+                        }}
+                        // className={classes.temporaryDrawer}
+                        classes={{ paper: classes.temporaryPaper, root: classes.drawerRoot }}>
+                        <Navigation />
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:168]: So I found that on the small screen the state of Navigation still lost. Maybe we should use context to preserve that?

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 7]
```diff
@@ -22,44 +87,89 @@ const useStyles = makeStyles((theme) => ({
...
+                        onClose={toggleDrawer}
+                        BackdropProps={{ invisible: true }}
+                        PaperProps={{ elevation: 0 }}
+                        variant="temporary"
+                        ModalProps={{
+                            BackdropProps: {
+                                classes: { root: classes.drawerBackdropRoot },
+                            },
+                        }}
+                        classes={{ paper: classes.temporaryPaper, root: classes.drawerRoot }}>
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:169]: Notice when you're changing a [deeper element](https://next.material-ui.com/guides/interoperability/#deeper-elements-3), the default className values like `.MuiSlider-thumb` does not working in our project. You may keep it for now too. I still not figure out what's the best practise for this style.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/components/DashboardFrame/Logo.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+                        id="形状结合"
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/Logo.tsx:24]: Hi please clean up those useless part in this SVG file thanks!

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 9]
```diff
@@ -1,62 +1,73 @@
...
 import { Routes } from '../../pages/routes'
+import { DashboardContext } from './context'
+import Logo from './Logo'
 
-const useStyle = makeStyles((theme: Theme) => ({
-    selected: {
+const ListItemLink = styled(({ nested, ...props }: LinkProps & ListItemProps & { nested?: boolean; to: string }) => (
+    <MuiListItem button component={Link} selected={!!useRouteMatch(props.to)} {...props} />
+))(({ theme, nested }) => ({
+    '&.MuiListItem-root': {
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:25]: I'm sorry maybe I'm not clear on my previous review. We should avoid specifying the global Material UI class names for now (in this case, `MuiListItem-root`). I have drafted my reason https://github.com/mui-org/material-ui/issues/20012#issuecomment-769642156 and requesting suggestions from the library team in the link.

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx [hunk 10]
```diff
@@ -1,62 +1,73 @@
...
-    )
-}
 
 export interface NavigationProps {}
 export function Navigation({}: NavigationProps) {
-    const classes = useStyle()
-    const [expanded, setExpanded] = useState(true)
+    const { expanded, toggleNavigationExpand } = useContext(DashboardContext)
+
+    const matches = useMediaQuery<Theme>((theme) => theme.breakpoints.up('lg'))
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/Navigation.tsx:55]: Maybe rename this to `isLargeScreen`?

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 11]
```diff
@@ -1,71 +1,195 @@
...
+const LeftContainer = styled(Grid)(({ theme }) => ({
+    height: '100vh',
+    [theme.breakpoints.up('lg')]: {
+        minWidth: 232,
     },
-    container: { backgroundColor: theme.palette.background.paper },
 }))
+
+const RightContainer = styled(Grid)(({ theme }) => ({
+    flex: 1,
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:33]: If you're not using the theme, you can use the tagged template form.

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 12]
```diff
@@ -1,71 +1,195 @@
...
+}))
+
 export interface DashboardFrameProps extends React.PropsWithChildren<{}> {}
+
 export function DashboardFrame(props: DashboardFrameProps) {
-    const classes = useStyles()
-    const menuStyle = useMediaQuery<Theme>((theme) => theme.breakpoints.down('md'))
+    const matches = useMediaQuery<Theme>((theme) => theme.breakpoints.down('lg'))
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:39]: Maybe rename this to isLargeScreen?

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 13]
```diff
@@ -1,71 +1,195 @@
...
 
+const MaskLogo = styled(Grid)`
+    flex-basis: 212px;
+    max-width: 212px;
+    & > svg {
+        flex: 1;
+    }
+`
+
+const Toolbar = styled((props: ToolbarProps) => <MuiToolbar {...props} />)(({ theme }) => ({
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:73]:
```suggestion
const Toolbar = styled(MuiToolbar)(({ theme }) => ({
```

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 14]
```diff
@@ -1,71 +1,195 @@
...
+const PageTitle = styled(Grid)(({ theme }) => ({
+    minHeight: 40,
+    alignItems: 'center',
+    paddingLeft: theme.spacing(4.25),
+    [theme.breakpoints.down('lg')]: {
+        flex: 1,
+    },
+}))
+
+const Stuff = styled('div')`
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:100]: For this very simple element only used once, you may want to use 

```js
<Box sx={{ flex: 1 }} />
```

or 

```js
<div css={css`flex: 1`} />
```

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 15]
```diff
@@ -1,71 +1,195 @@
...
 export interface PageFrameProps extends React.PropsWithChildren<{}> {
     title: React.ReactNode | string
     primaryAction?: React.ReactNode
 }
+
 export function PageFrame(props: PageFrameProps) {
-    const classes = useStyles()
     const left = typeof props.title === 'string' ? <Typography variant="h6">{props.title}</Typography> : props.title
     const right = props.primaryAction
+    const matches = useMediaQuery<Theme>((theme) => theme.breakpoints.down('lg'))
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:151]: Maybe rename this to isLargeScreen?

File: packages/dashboard/src/components/DashboardFrame/Navigation.tsx (hunk 9 above)
Comment (Jack-Works) [reply packages/dashboard/src/components/DashboardFrame/Navigation.tsx:25]: You can keep this style of code temporally as-is. I'll wait for the team to respond.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

**Review by Jack-Works — APPROVED** [review]
We got some potential problems with the styled component but it is safe in the dashboard. I will defer the problem to the future, wait for the team to responses or wrote an emotion-based makeStyles.

**Pull request merged**